### ISSUE TITLE
New Regions for Heroku Common Runtime

### ISSUE THREAD

**Issue by elimchaysengSF** [heroku/roadmap#195]
### Required Terms

- [X] I agree to follow this project's [Code of Conduct](https://github.com/heroku/roadmap/blob/main/CODE_OF_CONDUCT.md)
- [X] I have read and accept the [Salesforce Program Agreement](https://www.salesforce.com/company/program-agreement/)

### What service(s) is this request for?

Heroku dynos, Heroku Data, Common Runtime

### Tell us about what you're trying to solve. What challenges are you facing?

This year, we've started Heroku [Regional Expansion with our Private Spaces Product for Mumbai, Montreal](https://blog.heroku.com/private-spaces-expand-to-mumbai-and-montreal), Singapore, and London (with the full GA date later this year). New regions for the common runtime have been something internally we've been eager to enable. Now, we've also seen a significant amount of interest from customers for new online regions now that started including more for the Private Spaces product. 

We are currently in the research phase to add these regions, but I would like to hear from any customers about which specific regions they would like brought to the common runtime. Currently, there are only options to run your dynos in the US (us-east-1) and EU (eu-west-1), we are working to have parity across both offerings, but in order to help the sequencing of priority of which region we open first, please comment in the issue here and feel free to tag me. 

For reference, here are the current Private Spaces regions (with the additional that will be added in 2023). We will ideally start with a region from this list. 

```
dublin     Dublin, Ireland          Private Spaces
frankfurt  Frankfurt, Germany       Private Spaces
montreal   Montreal, Canada         Private Spaces
mumbai     Mumbai, India            Private Spaces
oregon     Oregon, United States    Private Spaces
sydney     Sydney, Australia        Private Spaces
tokyo      Tokyo, Japan             Private Spaces
virginia   Virginia, United States  Private Spaces
london     London, United Kingdom   Private Spaces 
singapore  Singpore, Singapore      Private Spaces
```

**Comment by pateljoel** [comment]
@elimchaysengSF I would say either Sydney, Singapore or Mumbai. The service I am running on Heroku has very high latency numbers in the eastern part of the world on the common runtime.

**Comment by elimchaysengSF** [comment]
Yep excellent points, will for sure take into consideration - thanks! @pateljoel

**Comment by manuel-reil** [comment]
Sydney - for the same above latency problems we are facing.

**Comment by caiocclopes** [comment]
How about Sao Paulo?

**Comment by etillier** [comment]
Canada

**Comment by raoulalwani** [comment]
+1 for Singapore, Tokyo, or Sydney. Any of those locations will vastly improve latency for users in Asia / Australasia compared to the existing options.

**Comment by wongpeiyi** [comment]
+1 for Singapore. The 200ms latency with EU or US is causing us to explore alternative providers like fly.io which has a Singapore region. The jump in cost to getting a private space just to address the latency is also not ideal.

Is there an estimate for how long it would take for the new region to be available, if this moves beyond the research phase?

**Comment by davidwessman** [comment]
I propose `eu-north-1`, mainly due to restrictions of data storage.

**Comment by glenbray** [comment]
+1 for Sydney

**Comment by krishofer** [comment]
+ for Sydney!

**Comment by xqliu** [comment]
Would like to know, is there any real plan for this ? 

Thanks :) 

Has waiting for this for > 3 years 😢

**Comment by capeterson** [comment]
Sorry for the long wait on this one. Mixed news that it's very much one of our strategic goals to get regional parity between common runtime and private spaces, but we also have some issues with how much effort, cost, infra, etc it takes to spin up new common runtime regions today that we really have to fix before we launch new ones. We've been laying some of the groundwork for this already in the last ~six months, but realistically @elimchaysengSF and I won't have the full info to share a delivery ETA until early next year. 

We also do want to go beyond the currently supported private spaces regions once this is done - but for some regions like Sao Paulo with materially different underlying cost to serve we're going to have to do some extra work to make our usage and billing systems able to manage region-aware pricing. No ETA on that yet, but it's going to be a high priority once we get common runtime parity with current regions closer to done.

**Comment by esquarenews** [comment]
100% Sydney

**Comment by rickwassing** [comment]
+1 Sydney

**Comment by bcosgraverz** [comment]
+1 for Singapore, Sydney, or Tokyo (in that order of preference)

**Comment by etillier** [comment]
Any progress on this? Particularly Montreal?

**Comment by Govinda-Fichtner** [comment]
Any news on this? I am particularly interested in Frankfurt/Germany.

**Comment by raphjutras** [comment]
I’m starting to think we’ll already have moved to another provider by the time this becomes a reality. We need a Canadian/Montreal server due to data storage restrictions. It’s just that we haven’t yet been _forced_ to. It's not even gonna be a  choice..

**Comment by etillier** [comment]
> I’m starting to think we’ll already have moved to another provider by the time this becomes a reality. We need a Canadian/Montreal server due to data storage restrictions. It’s just that we haven’t yet been _forced_ to. It's not even gonna be a choice..

Yes, same for us. We hung on with hope for years, because of this thread. No answer even to my last enquiry, so we finally have moved on, not that hard really, and all is good.

**Comment by raoulalwani** [comment]
Hi folks, any updates on this? @capeterson is this still a roadmap item or has it been shelved? Been with Heroku 15 years and it would be nice to have some new regions!

**Comment by cpsocket** [comment]
+1 Sydney
